### ISSUE TITLE
False positive on arguments of constructor created by "attrs"

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#3492]
### Steps to reproduce

```py
import attr

class Base:
    def __init__(self):
        pass

@attr.s(auto_attribs=True)
class Sub(Base):
    value: int

print(Sub(123))
```

### Current behavior

PyLint reports about the last line:
```
E1121: Too many positional arguments for constructor call (too-many-function-args)
```

### Expected behavior

No error should be printed, as the generated constructor takes two arguments (`self` and `value`).

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.8.2 (default, Mar 05 2020, 18:58:42) [GCC]
```

### Maybe related

Other issues about support for the `attrs` library are #1698 and  #2983. Superficially this issue doesn't seem to be the same thing, but perhaps there is overlap when digging deeper.

**Comment by zhiltsov-max** [comment]
Another way to reproduce and an option to mitigate:

```python
from attrs import define, field

@define(auto_exc=False, init=False)
class BaseError(Exception):
    v: int = field(default=0, kw_only=True)

    def __init__(self, msg=None, *, v=None):
        super().__init__(msg)
        self.__attrs_init__(v=v or 0)

# Pylint will raise false positive warnings for derived classes,
# when __init__ is defined directly. Monkey-patching resolves the problem:
# setattr(BaseError, '__init__', BaseError._my__init__)

@define(auto_exc=False)
class Error1(BaseError):
    x: int
    y: int

print(Error1(1, 2)) # << here
```

pylint pylint_example.py
************* Module pylint_example
pylint_example.py:20:6: E1121: Too many positional arguments for constructor call (too-many-function-args)

Attrs 21.4.0, pylint 2.10.2. But it happened in earlier versions too.
